api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/ConceptTranslatorImpl.java

```diff
@@ -52,14 +54,21 @@ public CodeableConcept toFhirResource(@Nonnull Concept concept) {
 				continue;
 			}
 			
-			String mappingType = mapping.getConceptMapType().getName();
-			if (mappingType == null || !mappingType.equalsIgnoreCase("SAME-AS")) {
+			ConceptMapType mapType = mapping.getConceptMapType();
+			boolean sameAs = mapType.getUuid() != null && mapType.getUuid().equals(ConceptMapType.SAME_AS_MAP_TYPE_UUID);
+			sameAs = sameAs || (mapType.getName() != null && mapType.getName().equalsIgnoreCase("SAME-AS"));
+			if (!sameAs) {
```

Comment:
There is an OpenMRS core constant for the uuid of the "same-as" concept map type.  So we should compare against this.  We can also look to see if the name is "SAME-AS" as was happening before - seems safe.